pythonscript/bindings/builtins_binder/vector3.cpp

```diff
@@ -264,6 +264,20 @@ static mp_obj_t _make_new_vector3(const mp_obj_type_t *type, size_t n_args, size
 }
 
 
+static mp_obj_t _unary_op_vector3(mp_uint_t op, mp_obj_t in) {
+  auto self = static_cast<Vector3Binder::mp_godot_bind_t*>(MP_OBJ_TO_PTR(in));
+  switch (op) {
+    case MP_UNARY_OP_NEGATIVE:
+      self->godot_vect2.x = -self->godot_vect2.x + 0.0;
```

Comment:
Unary operator should not modify the object itself, otherwise side effect occurs:

```python
v = Vector3(1, 1, 1)
assert v.x == 1
minus_v = -v  # here v is modified by the unary function...
assert minus_v.x == -1
assert v.x == 1  # failure, v.x is now -1 !!!
```